drakrun/drakrun/draksetup.py

```diff
@@ -1004,6 +1004,79 @@ def get_minio_client(config):
     )
 
 
+@click.group(help="Manage VM raw memory pre-sample dumps")
+def memdump():
+    pass
+
+
+@memdump.command(name="export", help="Upload pre-sample raw memory dump to MinIO.")
+@click.option("--instance", required=True, type=int, help="Instance ID of restored VM")
+@click.option(
+    "--bucket",
+    default="presample-memdumps",
+    help="MinIO bucket to store the compressed raw image",
+)
+def memdump_export(bucket, instance):
+    install_info = InstallInfo.try_load()
+    if install_info is None:
+        logging.error(
+            "Missing installation info. Did you forget to set up the sandbox?"
+        )
+        return
+
+    backend = get_storage_backend(install_info)
+    vm = VirtualMachine(backend, instance)
+    if vm.is_running:
+        logging.exception(f"vm-{instance} is running")
+        return
+
+    logging.info("Calculating snapshot hash...")
+    snapshot_hash = hashlib.sha256()
+    with open(os.path.join(VOLUME_DIR, "snapshot.sav"), "rb") as f:
+        for block in iter(lambda: f.read(65536), b""):
+            snapshot_hash.update(block)
+    snapshot_sha256 = snapshot_hash.hexdigest()
+    name = f"{snapshot_sha256}_pre_sample.raw_memdump.gz"
+
+    mc = get_minio_client(conf)
+
+    if not mc.bucket_exists(bucket):
+        logging.error("Bucket %s doesn't exist", bucket)
+        return
+
+    try:
+        mc.stat_object(bucket, name)
+        logging.info("This file already exists in specified bucket")
+        return
+    except NoSuchKey:
+        pass
+    except Exception:
+        logging.exception("Failed to check if object exists on minio")
+
+    logging.info("Restoring VM and performing memory dump")
+
+    try:
+        vm.restore()
```

Comment:
```suggestion
        vm.restore(pause=True)
```

Adding pause is quite important. Otherwise the VM will be running and the hash would be different.

Reply:
You are right, I neglected that fact due to `vmi-dump-memory` pausing VM by default, but I agree it's quite important